### PR TITLE
:gear: macos build for openssl

### PR DESCRIPTION
**Pull request by daminetreg**
This PR introduces a working macos build for openssl ( namely providing the correct sdkroot ).

Builds with :`tipi . -t clang-cxx17`

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -6,12 +6,19 @@ project(middleware_openssl VERSION "0.0.1")
...
 set(OPENSSL_SOURCE_DIR ${CMAKE_CURRENT_BINARY_DIR}/openssl-src)
 set(OPENSSL_INSTALL_DIR ${CMAKE_INSTALL_PREFIX})
+set(OPENSSLDIR /etc/ssl)
 set(OPENSSL_INCLUDE_DIR ${OPENSSL_INSTALL_DIR}/include)
 set(OPENSSL_CONFIGURE_COMMAND ${OPENSSL_SOURCE_DIR}/config)
 
+message("Machine is : ${CMAKE_SYSTEM_PROCESSOR} and ${CMAKE_C_FLAGS} ") 
+
 set(configure_command
+    #SDKROOT=${SDKROOT_PATH}
```
Comment (Lambourl) [inline CMakeLists.txt:16]: If it's only a comment, you can remove this line.

**Review by Lambourl — APPROVED** [review]
LGTM 👍

**Review by pysco68 — APPROVED** [review]
Even works on `-t linux-cxx17` 👍

**Comment by daminetreg** [comment]
> Even works on `-t linux-cxx17` 👍

Thanks, that's great to know ! I think we should also have CI for this one directly at some point.

**Pull request merged**